### DISCUSSION TITLE
Brightness reset to 0 on turning dimmer off through Z2M [solved]

### DISCUSSION THREAD

**Discussion by Saviq** [Koenkk/zigbee2mqtt#11091] [Devices]
Hey all, I'm using [the Lonsonho in-wall dimmers](https://www.zigbee2mqtt.io/devices/QS-Zigbee-D02-TRIAC-2C-LN.html) and all seemed fine until recently. When I turn a light off via Z2M, it gets the brightness set to 0 as well and this means the next time I try to turn it on from the physical switch, it turns on at 0 brightness.

```
Debug Received MQTT message on 'zigbee2mqtt/Salon/set' with data '{"state_l2":"OFF"}'
Debug Publishing 'set' 'state' to 'Salon'
Debug Received Zigbee message from 'Salon', type 'attributeReport', cluster 'genOnOff', data '{"onOff":0}' from endpoint 2 with groupID 0
Info MQTT publish: topic 'zigbee2mqtt/Salon', payload '{"brightness_l1":0,"brightness_l2":127,"linkquality":51,"state_l1":"OFF","state_l2":"OFF"}'
Info MQTT publish: topic 'zigbee2mqtt/Salon/l1', payload '{"brightness":0,"state":"OFF"}'
Info MQTT publish: topic 'zigbee2mqtt/Salon/l2', payload '{"brightness":127,"state":"OFF"}'
Info MQTT publish: topic 'zigbee2mqtt/Salon', payload '{"brightness_l1":0,"brightness_l2":127,"linkquality":48,"state_l1":"OFF","state_l2":"OFF"}'
Info MQTT publish: topic 'zigbee2mqtt/Salon/l1', payload '{"brightness":0,"state":"OFF"}'
Info MQTT publish: topic 'zigbee2mqtt/Salon/l2', payload '{"brightness":127,"state":"OFF"}'
```

So far so good, but after a second:

```
Debug Received Zigbee message from 'Salon', type 'attributeReport', cluster 'genLevelCtrl', data '{"currentLevel":0}' from endpoint 2 with groupID 0
Info MQTT publish: topic 'zigbee2mqtt/Salon', payload '{"brightness_l1":0,"brightness_l2":0,"linkquality":54,"state_l1":"OFF","state_l2":"OFF"}'
Info MQTT publish: topic 'zigbee2mqtt/Salon/l1', payload '{"brightness":0,"state":"OFF"}'
Info MQTT publish: topic 'zigbee2mqtt/Salon/l2', payload '{"brightness":0,"state":"OFF"}'
```
It's unclear to me what causes the last bit. I've been using this setup since May and I've only noticed this behaviour recently. I'm mostly using the physical switches… could I really have missed this before?

---

When turning off via physical switch:
```
Debug Received Zigbee message from 'Salon', type 'attributeReport', cluster 'genOnOff', data '{"onOff":0}' from endpoint 2 with groupID 0
Info MQTT publish: topic 'zigbee2mqtt/Salon', payload '{"brightness_l1":0,"brightness_l2":127,"linkquality":42,"state_l1":"OFF","state_l2":"OFF"}'
Info MQTT publish: topic 'zigbee2mqtt/Salon/l1', payload '{"brightness":0,"state":"OFF"}'
Info MQTT publish: topic 'zigbee2mqtt/Salon/l2', payload '{"brightness":127,"state":"OFF"}'
```
So no `brightness: 0` follow-up.

---

**EDIT**: after realizing only a few devices were affected, I tried removing them - and re-adding them fixed this for me. All behaving normally again.

**Comment by JHeijting** [comment]
Hello Saviq, did you encounter this again after re-adding the devices? I have exactly the same issue and have tried re-adding them a few times. Unfortunately this doesn't help.

Could you explain to me what you did exactly? At this moment I have the devices added to Z2M. To re-add them, I set Z2M to Join (all). Than I reset the dimmers by first having the light off and than press the switch for 10 seconds. In Z2M frontend I will than see that the device left the network and shortly after it rejoins the network. Unfortunately this does not result to a fix for me.

**Comment by Saviq** [comment]
@JHeijting that's really it, yeah, maybe remove them in Z2M first? Maybe that clears some state that causes this.

**Reply by JHeijting** [reply]
Thank you for your fast reply! I will try this.

**Reply by JHeijting** [reply]
I have deleted the devices using Z2M frontend and opted for the 'force delete' (otherwise it would automatically rejoin somehow). Than I have re-add them by setting join is True for coordinator only. Now it appears that I cannot reproduce the issue anymore. So fingers crossed for this being the fix!

This still keeps me thinking why it was caused to begin with. The first thing that pops up is that I might have been playing around with the specific feature 'transition'. In the end I put this back to 0 (so it should be disabled), but there might be something still causing the brightness going to 0 when using a Home Assistant Entity. I don't dare to reproduce this at this point though... The dimmers are used operational. But, I have some extra dimmers laying around that I have not found the time for to use operational, so I will soon make a test setup with one of them to try to reproduce this.

**Reply by Saviq** [reply]
> This still keeps me thinking why it was caused to begin with. The first thing that pops up is that I might have been playing around with the specific feature 'transition'.

Hmm now I think of it, it may have been that indeed. I was experimenting with dimming the light out in an automation, never got it to work well, but that may well have triggered this.
